pkg/workloadmeta/collectors/internal/docker/image_sbom_trivy.go

```diff
@@ -0,0 +1,162 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker && trivy
+// +build docker,trivy
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/trivy"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta/telemetry"
+
+	"github.com/docker/docker/client"
+)
+
+// scan buffer needs to be very large as we cannot block containerd collector
+const (
+	imagesToScanBufferSize = 5000
+)
+
+func imageMetadataCollectionIsEnabled() bool {
+	return config.Datadog.GetBool("container_image_collection.metadata.enabled")
+}
+
+func sbomCollectionIsEnabled() bool {
+	return imageMetadataCollectionIsEnabled() && config.Datadog.GetBool("container_image_collection.sbom.enabled")
+}
+
+func (c *collector) startSBOMCollection(ctx context.Context) error {
+	if !sbomCollectionIsEnabled() {
+		return nil
+	}
+
+	var err error
+	enabledAnalyzers := config.Datadog.GetStringSlice("container_image_collection.sbom.analyzers")
+	trivyConfiguration := trivy.DefaultCollectorConfig(enabledAnalyzers, config.Datadog.GetString("container_image_collection.sbom.cache_directory"))
+	trivyConfiguration.ClearCacheOnClose = config.Datadog.GetBool("container_image_collection.sbom.clear_cache_on_exit")
+	trivyConfiguration.DockerAccessor = func() (client.ImageAPIClient, error) {
+		return c.dockerUtil.RawClient(), nil
+	}
+
+	c.trivyClient, err = trivy.NewCollector(trivyConfiguration)
```

Comment:
Having a new collector implies that we're potentially creating a second Trivy instance, and a second cache instance, with the same parameters. There are at least two things to check:
- No race/deadlock/similar created by accessing the cache file in parallel
- No multiple scans in parallel.

These are the same issues as described in https://github.com/DataDog/datadog-agent/pull/15686 (which will conflict with this PR), though a lot less frequent in this case as it's les likely to have both container runtimes used on a host.